swift_targets.bzl

```diff
@@ -0,0 +1,251 @@
+def swift_get_compile_options(
+        exceptions = None,
+        rtti = None,
+        warning = None,
+        add_compile_options = None,
+        remove_compile_options = None,
+        cxx_standard = None,
+        c_standard = None):
+    if add_compile_options == None:
+        add_compile_options = []
+    if remove_compile_options == None:
+        remove_compile_options = []
+
+    for flag in add_compile_options + remove_compile_options:
+        if flag == "-fexceptions":
+            fail("Do not specify -fexceptions directly, use 'exceptions' argument instead")
+        if flag == "-fno-exceptions":
+            fail("Do not specify -fno-exceptions directly, avoid 'exceptions' argument instead")
+        if flag == "-frtti":
+            fail("Do not specify -frtti directly, use 'rtti' argument instead")
+        if flag == "-fno-rtti":
+            fail("Do not specify -fno-rtti directly, avoid 'rtti' argument instead")
+        if flag == "-Wno-error":
+            fail("Do not specify -Wno-error directly, use 'warning' to disable -Werror")
+
+    all_flags = []
+
+    # if not warning:
+    #     all_flags += ["-Werror", "-Wno-error=deprecated-declarations"]
```

Comment:
I had to disable `-Werror` because compiler complains when using C flag for C++ and vice versa.